lib/Bridge/TolerantParser/WorseReflection/KeywordCompletor.php

```diff
@@ -0,0 +1,136 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Phpactor\Completion\Bridge\TolerantParser\WorseReflection;
+
+use Generator;
+use Microsoft\PhpParser\Node;
+use Microsoft\PhpParser\Node\ClassMembersNode;
+use Microsoft\PhpParser\Node\Expression\AnonymousFunctionCreationExpression;
+use Microsoft\PhpParser\Node\Expression\MemberAccessExpression;
+use Microsoft\PhpParser\Node\Expression\ScopedPropertyAccessExpression;
+use Microsoft\PhpParser\Node\Expression\Variable;
+use Microsoft\PhpParser\Node\Statement\CompoundStatementNode;
+use Microsoft\PhpParser\Node\StringLiteral;
+use Microsoft\PhpParser\TokenStringMaps;
+use Phpactor\Completion\Bridge\TolerantParser\TolerantCompletor;
+use Phpactor\Completion\Core\Suggestion;
+use Phpactor\TextDocument\ByteOffset;
+use Phpactor\TextDocument\TextDocument;
+use function get_class;
+
+class KeywordCompletor implements TolerantCompletor
+{
+    public const MEMBER_ACCESS = "member_access";
+    public const CLASS_MEMBERS = "class_members";
+    public const STRING_LITERAL = "string_literal";
+    public const VARIABLE = "variable";
+    public const AFTER_ANONYMOUS_FUNC_PARAMS = "after_anonymous_func_params";
+
+    public const SPECIAL_SCOPES = [
+        self::MEMBER_ACCESS => [ ],
+        self::STRING_LITERAL => [ ],
+        self::VARIABLE => [ ],
+        // https://github.com/php/php-langspec/blob/master/spec/10-expressions.md#anonymous-function-creation
+        self::AFTER_ANONYMOUS_FUNC_PARAMS => [
+            'use'
+        ],
+        // https://github.com/php/php-langspec/blob/master/spec/14-classes.md#grammar-class-member-declaration
+        self::CLASS_MEMBERS => [
+            // visibility
+            'public',
+            'private',
+            'protected',
+            // scope
+            'static',
+            // property
+            'var',
+            // const
+            'const',
+            // method
+            'function',
+            // trait use
+            'use',
+        ],
+    ];
+
+    /**
+     * @var array
+     */
+    private static $allKeywords = null;
+
+    public function __construct()
+    {
+        if (self::$allKeywords === null) {
```

Comment:
can remove this. the class only gets instantiated once.

Reply:
could also make this lazy rather than being in the constructor - as eager initialization can have a cost if the class isn't used but it's added to an aggregate which would decide if it will be called or not.

In this case if the completor is disabled it won't be instantiated 

https://github.com/phpactor/completion-worse-extension/blob/24078fcb4d7956df4ce5d45199733d1f42f3ebc0/lib/CompletionWorseExtension.php#L130-L139

but maybe a good practice anyway (sure it doesn't make much diff here)